sdk/iOS/OoyalaSkinSDK/OOUpNextManager.m

```diff
@@ -82,8 +84,37 @@ - (void)goToNextVideo {
   // if upnext has not been dismissed and there is a next video, play the next video.
   LOG(@"Going to next video based on Up Next");
   if (!self.isDismissed && self.nextVideo) {
-    [self.player setEmbedCode:self.nextVideo[embedCodeKey]];
+    
+    __weak typeof(self) weakSelf = self;
+    NSString *expectedEmbedCode = weakSelf.nextVideo[embedCodeKey];
+    void (^expectedBlock) (OOVideo *currentItem); //params of OOCurrentItemChangedCallback
+    expectedBlock = ^(OOVideo *currentItem) {
+      LOG(@"Method -goToNextVideo got expectedBlock");
+      NSString *arrivedCode = currentItem.embedCode;
+      
+      //OS: must be checked, because OOBaseStreamPlayer observes 'AVPlayerItemStatusReadyToPlay' two times: when asset just attached and when asset is completed.
+      if ([arrivedCode isEqualToString:expectedEmbedCode]) {
+        LOG(@"SUCCESS: asset with expected embed code [%@] ", arrivedCode);
+        [weakSelf.player play];
+        // If success - block 'currentItemChangedCallback' must be removed after '[weakSelf.player play]', to prevent ignition from OOBaseStreamPlayer's KVO 'AVPlayerItemStatusReadyToPlay'
+        weakSelf.player.currentItemChangedCallback = nil;
+      } else {
+        LOG(@"❌ player with embed code [%@] that is not expected", arrivedCode);
+      }
+    };
+    self.player.currentItemChangedCallback = expectedBlock;
   }
+  
+#warning: old API. Remove when SDK version > 4.46.0_GA. If you need to use old API, uncomment -playForJustChangedItem in -bridgeCurrentItemChangedNotification: of 'OOSkinPlayerObserver'
+    //[self.player setEmbedCode:self.nextVideo[embedCodeKey]];
+    
+    //new API from OOyalaSDK. Available > 4.46.0_GA
+    [self.player setEmbedCode:self.nextVideo[embedCodeKey] withCallback:^(OOOoyalaError *error) {
+      LOG(@"goToNextVideo setEmbedCode got callback");
+      if (error) {
+        LOG(@"❌ SetEmbedCode is NOT successfull. Error: %@", error.debugDescription);
```

Comment:
I don't think we need to print the error itself:
- OoyalaSDK prints all errors
- OoyalaSkinSDK displays these errors in the app

Reply:
Done